app/code/Magento/InventoryAdminUi/view/adminhtml/ui_component/inventory_source_form.xml

```diff
@@ -183,20 +183,22 @@
                 </select>
             </formElements>
         </field>
-        <field name="region_id" formElement="select" sortOrder="20">
+        <field name="region_id"
+               component="Magento_InventoryAdminUi/js/form/element/region"
+               formElement="select" sortOrder="20">
             <settings>
                 <dataType>text</dataType>
                 <label translate="true">State/Province</label>
             </settings>
             <formElements>
                 <select>
                     <settings>
-                        <options class="Magento\InventoryAdminUi\Model\OptionSource\RegionSource"/>
                         <filterBy>
                             <field>country_id</field>
                             <target>${ $.provider }:${ $.parentScope }.country_id</target>
                         </filterBy>
                         <customEntry>region</customEntry>
+                        <options class="Magento\Directory\Model\ResourceModel\Region\Collection"/>
```

Comment:
Why we need these changes?
**\Magento\InventoryAdminUi\Model\OptionSource\RegionSource** looks like enough to provide list of regions + it has run-cache.

Reply:
Changed